v3/lints/cabf_smime_br/lint_adobe_extensions_legacy_multipurpose_criticality.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_smime_br
+
+import (
+	"github.com/zmap/zcrypto/x509"
+
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_adobe_extensions_legacy_multipurpose_criticality",
+			Description:   "If present, Adobe Time‐stamp X509 extension (1.2.840.113583.1.1.9.1) or the Adobe ArchiveRevInfo extension (1.2.840.113583.1.1.9.2) SHALL NOT be marked as critical for multipurpose/legacy SMIME certificates",
+			Citation:      "7.1.2.3.m",
+			Source:        lint.CABFSMIMEBaselineRequirements,
+			EffectiveDate: util.CABF_SMIME_BRs_1_0_0_Date,
+		},
+		Lint: NewAdobeExtensionsLegacyMultipurposeCriticality,
+	})
+}
+
+type adobeExtensionsLegacyMultipurposeCriticality struct{}
+
+// NewAdobeExtensionsLegacyMultipurposeCriticality creates a new linter to enforce adobe x509 extensions requirements for multipurpose or legacy SMIME certs
+func NewAdobeExtensionsLegacyMultipurposeCriticality() lint.LintInterface {
```

Comment:
```suggestion
func NewAdobeExtensionsLegacyMultipurposeCriticality() lint.CertificateLintInterface {
```
Per #699, I think LintInterface is also deprecated.